### PR TITLE
Checks if MSVS 2019 or 2022 exists through reg key

### PR DESCRIPTION
**Pull request by SlanDF**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

Ive added some changes to the .ps1 script that the system dependencies are at the correct version. In this case I've made the validation for the MSVC to be made using the path inside the registry to make sure we have the correct path no matter where MSVS is installed. I fetch the directory from there then travel to that directory, pop the path once and that places you in the folder that contains the version of the MSVST that you want. Then it checks if the 2019 or 2022 folders are there, if they are then it passes the validation. I kept the same error handling that was there before, and only commented the previous code that handled this part of the script. I also added error handling in case the registry path to vs doesnt exist. 

#### Motivation and Context

The way it was previously implemented didnt really made much sense, and i was having issue passing that validation knowing fully well i had my MSVST updated. So I went and digged on how to find this version you wanted keeping in consideration it can be installed anywhere (hence the reg key solution)
I honestly think this could even be changed to go by version number because thats was more easy to access in the registry and you could just set if the version>14 or something

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

#### Release Notes

none

### CODE REVIEW

File: scripts/build-windows.ps1 [hunk 1]
```diff
@@ -94,16 +93,37 @@ if ($env:CLEAN -eq "true")
...
+}
+
+if(-not $ACTUAL_MSVST_VERSION){
   fail_with_docs "You are not running the expected version of MSVS Tools!
-    expected: [$EXPECTED_MSVST_VERSION]
-    actual  : [$ACTUAL_MSVST_VERSION]"
+        expected: [$EXPECTED_MSVST_VERSION]"
 }
 
+# -------------------------LEGACY----------------------------------
```
Comment (vraravam) [inline scripts/build-windows.ps1]: Please remove this commented out portion. If needed, we can revert the commit to resurrect it

**Review by vraravam — CHANGES_REQUESTED** [review]

**Comment by vraravam** [comment]
Also - in the new commit that you are going to do, please add `[skip ci]` at the end of the commit message.
Reasoning: This change has no direct impact on the software being built. Moreover, its only for local builds on contributors machines - and thus we can avoid triggering a CI build as its unnecessary.

**Comment by SpecialAro** [comment]
image
@Kanubiz this is currently failing on my system. I can assure it was well on the other script.

**Comment by vraravam** [comment]
@SpecialAro - what's the version that's printed if you run that line manually in your system?

**Comment by SlanDF** [comment]
Me and @SpecialAro figured a better way to check the actual version
`Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\14.0\VC\Runtimes\X64` Includes the version number that you want to really check
v14.20 to v14.29 is 2019
v14.30 to v14.32 (latest) is 2022
like this we just need to go to that reg path then assign ` $ACTUAL_MSVST_VERSION` to either 2019 or 2022 based on the above conditions

**Pull request merged**